### PR TITLE
Add CI to check formatting and run static analysis

### PR DESCRIPTION
**Pull request by auscompgeek**

### CODE REVIEW

**Comment by auscompgeek** [comment]
Commit statuses can be seen at https://github.com/auscompgeek/pyinfiniterecharge/compare/master...ci-lint

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
```
Comment (raulvera59) [inline .github/workflows/check.yml]: I thought we'd standardised on 3.8?

Comment (auscompgeek) [reply .github/workflows/check.yml]: If we really want to use 3.8-specific syntax we can run pyflakes on 3.8. But I don't imagine we will be using any syntax introduced in 3.8.

**Comment by raulvera59** [comment]
That's not really the point. The issue is that if we agreed to standardise
on a specific version, and I thought we did, then our tooling should
reflect that, or we didn't actually standardise.

And we should standardise, precisely to avoid mismatches that could bite us
at the last minute. Prefer impossible to unlikely.

**Comment by auscompgeek** [comment]
These are static analysis tools, not running the robot code. black doesn't care which version of Python 3.6+ it's running on, and pyflakes only cares for parsing its input. I'm more than happy to run tests in CI on Python 3.8 once that's feasible.

I would personally prefer if we didn't use Python 3.8 syntax in our robot code. I'm not aware of any discussion we've had to "standardise" on any particular Python 3.x version.

**Comment by raulvera59** [comment]
I could be mistaken, but I recall Lucien saying that we would be using
python 3.8, at one of the training sessions in October/November. That's why
I upgraded to 3.8 from 3.6 when setting up the VM and broke it.

As you point out it matters less for the version that the tools run on, but
I think we should stick to a single version for the robot code. Which
version doesn't matter, but sticking to one helps prevent a whole class of
hard-to-find bugs. I've seen whole teams stalled for days due to seemingly
innocuous version skew.

If you would prefer we didn't use anything from 3.8 then we should
standardise on 3.6 or 3.7. Let's pick one and tell everyone to ensure that
that's the version they are using for robot code.

**Pull request merged**